rig/machine_control/machine_controller.py

```diff
@@ -2110,6 +2132,19 @@ def chips(self):
         """
         return iter(self)
 
+    def ethernet_connected_chips(self):
+        """Iterate over the coordinates of Ethernet connected chips.
+
+        Yields
+        ------
+        ((x, y), str)
+            The coordinate and IP address of each Ethernet connected chip in
+            the system.
+        """
+        return ((xy, chip_info.ip_address)
```

Comment:
Presumably you mean yielding `(xy, chip_info.ip_address)` rather than changing this to `(xy, ChipInfo)`? I had considered the latter but couldn't think of a good use case.